src/stan/variational/families/normal_fullrank.hpp

```diff
@@ -49,12 +49,15 @@ namespace stan {
 
     public:
       // Constructors
+
+      // Constructs a zero-filled variational distribution
       explicit normal_fullrank(size_t dimension) :
         dimension_(dimension) {
         mu_     = Eigen::VectorXd::Zero(dimension_);
-        L_chol_  = Eigen::MatrixXd::Identity(dimension_, dimension_);
+        L_chol_ = Eigen::MatrixXd::Zero(dimension_, dimension_);
```

Comment:
Arguments should be declared `const` wherever possible.  For instance, the model m in `calc_grad` (at least I hope the functions are const enough in the model --- if not, let me know and I'll fix that --- you won't be able to fix it in this pull request). 

It would help if you provided `@param[in,out]` with in for when parameters must be defined coming in and out when they're modified.  Obviously, the in only ones should be constant (other than primitives, which don't need const declarations).